compiler/rustc_attr/src/builtin.rs

```diff
@@ -526,6 +525,26 @@ fn gate_cfg(gated_cfg: &GatedCfg, cfg_span: Span, sess: &ParseSess, features: &F
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+struct Version {
+    major: usize,
+    minor: usize,
+    patch: usize,
```

Comment:
If we are keeping "invalid version literal" an error, then it's probably better to use `u16` for all the fields to be conservative, that's also what `version_check` uses.

If we are not keeping this an error, then it's probably better to use `u64` (because why would the version predicate result be dependent on pointer size).

Reply:
Yeah good point, it shouldn't suddenly behave differently if the compiler gets built for 32 bit.